File: 3010-divide-an-array-into-subarrays-with-minimum-cost-i/3010-divide-an-array-into-subarrays-with-minimum-cost-i.cpp
class Solution {
public:
    int minimumCost(vector<int>& nums) {
        int ans = nums[0];
        int firstMin = 51;
        int secMin = 51;
        for (int i = 1; i < nums.size(); i++) {
            int num = nums[i];
            if (num < firstMin) {
                secMin = firstMin;
                firstMin = num;
            }
            else if (num < secMin) {
                secMin = num;
            }
        }
        return ans + firstMin + secMin;
    }
};

File: 3010-divide-an-array-into-subarrays-with-minimum-cost-i/3010-divide-an-array-into-subarrays-with-minimum-cost-i.py
class Solution:
    def minimumCost(self, nums: List[int]) -> int:
        ans = nums[0]
        first_min = 51
        second_min = 51

        for num in nums[1:]:
            if num < first_min:
                second_min = first_min
                first_min = num
            elif num < second_min:
                second_min = num

        return ans + first_min + second_min

        